app/controllers/admin/users_controller.rb

```diff
@@ -40,6 +40,30 @@ def update
       redirect_to "/admin/users/#{params[:id]}"
     end
 
+    def destroy
+      role = params[:role].to_sym
```

Comment:
**suggestion (non-blocking):** Time for a little Rails trick 😃 Have you ever wondered how things like `Rails.env.production?` work? The answer is a helpful little class called [`ActiveSupport::StringInquirer`](https://api.rubyonrails.org/classes/ActiveSupport/StringInquirer.html). `String` also has an [inquiry method](https://www.rubydoc.info/gems/activesupport/String:inquiry) which turns any string into an instance of this class:

```ruby
role = params[:role].inquiry
role.single_resource_admin?
#=> false
role.super_admin?
#=> true
```

If you do this your conditions read nicer and you don't have to manually convert it back to a string by calling `to_s` in the flash message generation:

```ruby
role.class
#=> ActiveSupport::StringInquirer
role.humanize.titlecase
#=> "Super Admin"
```

Reply:
This is so neat - TIL! 😄 The only issue that I'm running into is that when I change `role = params[:role].to_sym` to `role = params[:role].inquiry`, I run into an issue where _all_ `:single_resource_admin` roles are removed rather than the particular one that I'm trying to remove. I attempted to add `.to_sym` inline when removing a `:single_resource_admin` role, but it still removed all `single_resource_admin` roles rather than the particular one that I was trying to remove.

Reply:
Did you try to do that before or after adding the `nil` check?

Reply:
I _believe_ I tried it both before and after. I just tried using `.inquiry` in place of `.to_sym` again and ran into the same result. 😕

Reply:
That's curious. It really isn't all that important, the code is fine as-is. But if you want I can take a look.